samples/bluetooth/ccp_call_control_client/src/main.c

```diff
@@ -0,0 +1,335 @@
+{
+	if (err == 0) {
+		k_sem_give(&sem_security_updated);
+	} else {
+		LOG_ERR("Failed to set security level: %s(%u)", bt_security_err_to_str(err), err);
+	}
+}
+
+BT_CONN_CB_DEFINE(conn_callbacks) = {
+	.connected = connected_cb,
+	.disconnected = disconnected_cb,
+	.security_changed = security_changed_cb,
+};
+
+static bool check_gtbs_support(struct bt_data *data, void *user_data)
+{
+	struct net_buf_simple svc_data;
+	bool *connect = user_data;
+	const struct bt_uuid *uuid;
+	uint16_t uuid_val;
+
+	if (data->type != BT_DATA_SVC_DATA16) {
+		return true; /* Continue parsing to next AD data type */
+	}
+
+	if (data->data_len < sizeof(uuid_val)) {
+		LOG_WRN("AD invalid size %u", data->data_len);
+		return true; /* Continue parsing to next AD data type */
+	}
+
+	net_buf_simple_init_with_data(&svc_data, (void *)data->data, data->data_len);
+
+	/* Pull the 16-bit service data and compare to what we are searching for */
+	uuid_val = net_buf_simple_pull_le16(&svc_data);
+	uuid = BT_UUID_DECLARE_16(sys_le16_to_cpu(uuid_val));
+	if (bt_uuid_cmp(uuid, BT_UUID_GTBS) != 0) {
+		/* We are looking for the GTBS service data */
+		return true; /* Continue parsing to next AD data type */
+	}
+
+	*connect = true;
+
+	return false; /* Stop parsing */
+}
+
+static void scan_recv_cb(const struct bt_le_scan_recv_info *info, struct net_buf_simple *ad)
+{
+	char addr_str[BT_ADDR_LE_STR_LEN];
+	bool connect = false;
+
+	if (peer_conn != NULL) {
+		/* Already connected */
+		return;
+	}
+
+	/* CCP mandates that connectbale extended advertising is used by the peripherals so we
+	 * ignore any scan report this is not that.
+	 * We also ignore reports with poor RSSI
+	 */
+	if (info->adv_type != BT_GAP_ADV_TYPE_EXT_ADV ||
+	    (info->adv_props & BT_GAP_ADV_PROP_EXT_ADV) == 0 ||
+	    (info->adv_props & BT_GAP_ADV_PROP_CONNECTABLE) == 0 || info->rssi < -70) {
+		return;
+	}
+
+	(void)bt_addr_le_to_str(info->addr, addr_str, sizeof(addr_str));
+	LOG_INF("Connectable device found: %s (RSSI %d)", addr_str, info->rssi);
+
+	/* Iterate on the advertising data to see if claims GTBS support */
+	bt_data_parse(ad, check_gtbs_support, &connect);
+
+	if (connect) {
+		int err;
+
+		err = bt_le_scan_stop();
+		if (err != 0) {
+			LOG_ERR("Scanning failed to stop (err %d)", err);
+			return;
+		}
+
+		LOG_INF("Connecting to found CCP server");
+		err = bt_conn_le_create(info->addr, BT_CONN_LE_CREATE_CONN,
+					BT_LE_CONN_PARAM_DEFAULT, &peer_conn);
+		if (err != 0) {
+			LOG_ERR("Conn create failed: %d", err);
+		}
+	}
+}
+
+static int scan_and_connect(void)
+{
+	int err;
+
+	err = bt_le_scan_start(BT_LE_SCAN_PASSIVE, NULL);
+	if (err != 0) {
+		LOG_ERR("Scanning failed to start (err %d)", err);
+		return err;
+	}
+
+	LOG_INF("Scanning successfully started");
+
+	err = k_sem_take(&sem_conn_state_change, K_FOREVER);
+	if (err != 0) {
+		LOG_ERR("failed to take sem_connected (err %d)", err);
+		return err;
+	}
+
+	err = bt_conn_set_security(peer_conn, BT_SECURITY_L2);
```

Comment:
For real devices, would this also need BT_SECURITY_FORCE_PAIR?

Reply:
Not really - If we are already bonded we don't need to create a new pairing each time